x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -405,8 +405,8 @@ protected LogicalPlan doRule(LogicalPlan plan) {
                 childrenOutput.addAll(output);
             }
 
-            if (plan instanceof Stats stats) {
```

Comment:
By exposing the Aggregate, there's no reason to have the Stats interface anymore.
The upside is the Aggregate resolution gets reused automatically (including nested filters, etc..)